src/category_theory/concrete_category/basic.lean

```diff
@@ -40,15 +40,15 @@ namespace category_theory
 section prio
 set_option default_priority 100 -- see Note [default priority]
 /-- A concrete category is a category `C` with a fixed faithful functor `forget : C ⥤ Type`. -/
-class concrete_category (C : Type (u+1)) extends category.{u} C :=
+class concrete_category (C : Type (u+1)) [category.{u} C] :=
 (forget : C ⥤ Type u)
 [forget_faithful : faithful forget]
 end prio
 
 attribute [instance] concrete_category.forget_faithful
 
 /-- The forgetful functor from a concrete category to `Type u`. -/
-@[reducible] def forget (C : Type (u+1)) [concrete_category C] : C ⥤ Type u :=
+@[reducible] def forget (C : Type (u+1)) [category.{u} C] [concrete_category C] : C ⥤ Type u :=
```

Comment:
```suggestion
@[reducible] def forget (C : Type (u+1)) {cat : category.{u} C} [concrete_category C] : C ⥤ Type u :=
```
Should these now all be made implicit, because they can be inferred?

Reply:
Personally, I think this is premature optimisation. It makes the code meaningfully less readable:
1. you need to introduce arbitrary names
2. readers have to think about all the varieties of brackets
3. the trick only works about half the time in this file, so we'd need to add explanations about why we are and aren't doing it in various places. :-(

Reply:
Ok, if it only works half of the time, let's not do it.